File: forest/signal-corpse.clj
(while true
  (def green (.findFlag this "green"))
  (def black (.findFlag this "black"))
  (def enemy (.findNearest this (.findEnemies this)))
  (if green
    (.pickUpFlag this green)
    (if (and black (.isReady this "cleave"))
      (when true
        (.pickUpFlag this black)
        (def enemy (.findNearest this (.findEnemies this)))
        (.cleave this enemy))
      (if (and enemy (< (.distanceTo this enemy) 10))
        (if (.isReady this "bash")
          (.bash this enemy)
          (.attack this enemy))))))

File: forest/signal-corpse.lua
loop
    local enemy = self:findNearest(self:findEnemies())
    local green = self:findFlag("green")
    local black = self:findFlag("black")
    if green then
        self:pickUpFlag(green)
    elseif black and self:isReady("cleave") then
        self:pickUpFlag(black)
        enemy = self:findNearest(self:findEnemies())
        self:cleave(enemy)
    elseif enemy and self:distanceTo(enemy) < 10 then
        self:attack(enemy)
    end
end
